src/libs/TripReservationUtils.ts

```diff
@@ -105,25 +105,36 @@ function bookATrip(translate: LocaleContextProps['translate'], setCtaErrorMessag
         Navigation.navigate(ROUTES.WORKSPACE_PROFILE_ADDRESS.getRoute(activePolicyID ?? '-1', Navigation.getActiveRoute()));
         return;
     }
-    if (!travelSettings?.hasAcceptedTerms) {
-        Navigation.navigate(ROUTES.TRAVEL_TCS);
-        return;
-    }
-    if (ctaErrorMessage) {
-        setCtaErrorMessage('');
-    }
-    Link.openTravelDotLink(activePolicyID)
-        ?.then(() => {
-            if (!NativeModules.HybridAppModule || !isSingleNewDotEntry) {
-                return;
-            }
 
-            Log.info('[HybridApp] Returning to OldDot after opening TravelDot');
-            NativeModules.HybridAppModule.closeReactNativeApp(false, false);
-        })
-        ?.catch(() => {
-            setCtaErrorMessage(translate('travel.errorMessage'));
-        });
+    const isPolicyProvisioned = policy?.travelSettings?.spotnanaCompanyID || policy?.travelSettings?.associatedTravelDomainAccountID;
+    if (policy?.travelSettings?.hasAcceptedTerms || (travelSettings?.hasAcceptedTerms && isPolicyProvisioned)) {
+        Link.openTravelDotLink(activePolicyID)
+            ?.then(() => {
+                if (!NativeModules.HybridAppModule || !isSingleNewDotEntry) {
+                    return;
+                }
+
+                Log.info('[HybridApp] Returning to OldDot after opening TravelDot');
+                NativeModules.HybridAppModule.closeReactNativeApp(false, false);
+            })
+            ?.catch(() => {
+                setCtaErrorMessage(translate('travel.errorMessage'));
+            });
+        if (ctaErrorMessage) {
+            setCtaErrorMessage('');
+        }
+    } else if (isPolicyProvisioned) {
+        Navigation.navigate(ROUTES.TRAVEL_TCS.getRoute(CONST.TRAVEL.DEFAULT_DOMAIN));
+    } else {
+        const adminDomains = PolicyUtils.getAdminsPrivateEmailDomains(activePolicyID);
+        if (adminDomains.length === 0) {
+            Navigation.navigate(ROUTES.TRAVEL_PUBLIC_DOMAIN_ERROR);
```

Comment:
For this, it is slightly better to have a variable for which route to use, and then just make a single call to `Navigation.navigate(routeToNavigateTo)`

Reply:
fixed.